lib/InputWithButton/index.js

```diff
@@ -1,42 +1,60 @@
-import React from 'react';
+import React, { Component } from 'react';
 import cx from 'classnames';
 import PropTypes from 'prop-types';
 import Button from '../Button';
 
-const InputWithButton = ({
-  value,
-  buttonText,
-  onClick,
-  inputId,
-  buttonId,
-  disabled
-}) => (
-  <div className="input-with-button">
-    <input
-      readOnly
-      id={inputId}
-      aria-labelledby={buttonId}
-      value={value}
-      className={cx('input-with-button__input', {
-        'input-with-button__input--disabled': disabled
-      })}
-    />
-    <Button
-      id={buttonId}
-      disabled={disabled}
-      types={['primary', 'input-appendage']}
-      onClick={() => {
-        const inputElement = document.getElementById(inputId);
-        onClick(inputElement);
-      }}
-    >
-      {buttonText}
-    </Button>
-  </div>
-);
+class InputWithButton extends Component {
+  state = {
+    hasBeenClicked: false
+  };
+
+  render() {
+    const {
+      value,
+      buttonText,
+      buttonTextAfterClick,
+      onClick,
+      inputId,
+      buttonId,
+      disabled
+    } = this.props;
+
+    const { hasBeenClicked } = this.state;
+
+    return (
+      <div className="input-with-button">
+        <input
+          readOnly
+          id={inputId}
+          aria-labelledby={buttonId}
+          value={value}
+          className={cx('input-with-button__input', {
+            'input-with-button__input--disabled': disabled
+          })}
+        />
+        <Button
+          id={buttonId}
+          disabled={disabled}
+          types={['primary', 'input-appendage']}
+          onClick={() => {
```

Comment:
could move this out to its own onButtonClick function, but thats entirely personal preference